dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -869,24 +874,24 @@ public List<BlockWorkerInfo> getCachedWorkers() throws IOException {
    * @param type get worker list type
    * @return the info of all block workers eligible for reads and writes
    */
-  public List<BlockWorkerInfo> getCachedWorkers(GetWorkerListType type) throws IOException {
-    synchronized (mWorkerInfoList) {
-      if (mWorkerInfoList.get() == null || mWorkerInfoList.get().isEmpty()
+  public WorkerClusterView getCachedWorkers(GetWorkerListType type) throws IOException {
+    synchronized (mCachedWorkerClusterView) {
+      if (mCachedWorkerClusterView.get() == null || mCachedWorkerClusterView.get().isEmpty()
           || mWorkerRefreshPolicy.attempt()) {
         switch (type) {
           case ALL:
-            mWorkerInfoList.set(getAllWorkers());
+            mCachedWorkerClusterView.set(getAllWorkers());
```

Comment:
oh interestingly, this same cache object is caching results for 3 types? how do we know if a get wants the same type as is cached?

Reply:
good catch! I think this is a potential bug. Since the only caller to this overloaded `getCachedWorkers(GetWorkerListType)` is `DoraCacheClient.getWorkerNetAddress` and it's argument `GetWorkerListType` is tied to `mEnableDynamicHashRing` which is currently a runtime constant, this does not cause any visible bug.

IMO since there are already `getLiveWorkers` `getAllWorkers` and `getLostWorkers` methods on `FileSystemContext`, I'd propose to simply deprecate/remove this method.

Reply:
to fix this in a separate PR.

Reply:
good catch